src/node_internals.h

```diff
@@ -294,8 +294,35 @@ v8::MaybeLocal<v8::Value> InternalMakeCallback(
     v8::Local<v8::Value> argv[],
     async_context asyncContext);
 
+class InternalCallbackScope {
+ public:
+  enum ResourceExpectation { kRequireResource, kAllowEmptyResource };
```

Comment:
Can you add some code comments here that describe briefly the difference between these flags

Reply:
@jasnell done + rebased